lib/repl.js

```diff
@@ -585,10 +595,21 @@ REPLServer.prototype.createContext = function() {
     context = global;
   } else {
     context = vm.createContext();
-    for (var i in global) context[i] = global[i];
-    context.console = new Console(this.outputStream);
     context.global = context;
     context.global.global = context;
+    const _console = new Console(this.outputStream);
+    Object.defineProperty(context, 'console', {
+      configurable: true,
+      enumerable: true,
+      get: () => _console
+    });
+    Object.getOwnPropertyNames(global).filter((name) => {
+      if (name === 'console' || name === 'global') return false;
+      return GLOBAL_OBJECT_PROPERTY_MAP[name] === undefined;
```

Comment:
This has false positives for entries from `Object.prototype`, e.g. for `name === 'constructor'`. I’m not sure if that’s actually relevant, but this line should be read as equivalent to `GLOBAL_OBJECT_PROPERTIES.indexOf(name) === -1`, right?